src/bindgen/cargo/cargo_expand.rs

```diff
@@ -75,6 +75,11 @@ pub fn expand(
         cmd.env("CARGO_TARGET_DIR", _temp_dir.unwrap().path());
     } else if let Ok(ref path) = env::var("CARGO_EXPAND_TARGET_DIR") {
         cmd.env("CARGO_TARGET_DIR", path);
+    } else {
+       // Don't use the default dir, else we could end up in a deadlock when cbindgen is run from
+       // a build.rs file
+       let out_dir = PathBuf::from(&env::var("OUT_DIR").unwrap());
```

Comment:
Doesn't this break CLI usage? There's no guarantee that `OUT_DIR` exists when called from the CLI.

Can we instead check the return value of `env::var` and only set it if `OUT_DIR` exists?

Or am I missing something?

Reply:
I haven't thought about CLI use, you are right. I'll add this check.